File: BatmanCharacters/BatmanCharacter.m
#import "BatmanCharacter.h"

@implementation BatmanCharacter

+ (NSMutableArray *) initialSetOfBatmanCharacters {
    NSMutableArray *batmanCharacters = [[NSMutableArray alloc] init];
    
    BatmanCharacter *joker = [[BatmanCharacter alloc] init];
    joker.characterName = @"Joker";
    BatmanCharacter *riddler = [[BatmanCharacter alloc] init];
    riddler.characterName = @"Riddler";
    BatmanCharacter *twoFace = [[BatmanCharacter alloc] init];
    twoFace.characterName = @"TwoFace";
    BatmanCharacter *batman = [[BatmanCharacter alloc] init];
    batman.characterName = @"Batman";
    
    [batmanCharacters addObject:joker];
    [batmanCharacters addObject:riddler];
    [batmanCharacters addObject:twoFace];
    [batmanCharacters addObject:batman];
    
    return batmanCharacters;
}

@end

File: BatmanCharacters/BatmanCharacter.swift

import Foundation

@objc class BatmanCharacter : NSObject {
    
    var characterName:String
    var characterViewed:Bool
    
    class func initialSetOfBatmanCharacters() -> [BatmanCharacter] {
        var joker:BatmanCharacter = BatmanCharacter(characterName: "Joker")
        var riddler:BatmanCharacter = BatmanCharacter(characterName: "Riddler")
        var twoface:BatmanCharacter = BatmanCharacter(characterName: "TwoFace")
        var batman:BatmanCharacter = BatmanCharacter(characterName: "Batman")
        
        var batmanCharacters:[BatmanCharacter] = []
        batmanCharacters.append(joker)
        batmanCharacters.append(riddler)
        batmanCharacters.append(twoface)
        batmanCharacters.append(batman)
        
        return batmanCharacters
    }
    
    init(characterName:String) {
        self.characterName = characterName
        self.characterViewed = false
    }
    
}
